src/Models/HydrostaticFreeSurfaceModels/update_hydrostatic_free_surface_model_state.jl

```diff
@@ -28,6 +28,129 @@ they are called in the end.
 update_state!(model::HydrostaticFreeSurfaceModel, callbacks=[]; compute_tendencies = true) =
          update_state!(model, model.grid, callbacks; compute_tendencies)
 
+function fill_paired_halo_regions!(fields, signed=true)
+
+    field₁, field₂ = fields
+    grid = field₁.grid
+
+    if !(grid isa ConformalCubedSphereGrid)
+        return
+    end
+
+    Nx, Ny, Nz = size(grid)
+    Hx, Hy, Hz = halo_size(grid)
+    signed ? plmn = -1 : plmn = 1
+#- will not work if (Nx,Hx) and (Ny,Hy) are not equal
+    Nc = Nx ; Hc = Hx
```

Comment:
```suggestion

instead of a comment (which might not be seen), let's do

```suggestion

Nx == Ny || error("horizontal grid size Nx and Ny must be the same")
Nc = Nx

Hx == Hy || error("horizontal halo size Hx and Hy must be the same")
Hc = Hx
```

Reply:
Done. 🙂